src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs

```diff
@@ -104,6 +105,52 @@ internal void Skip()
             }
         }
 
+        internal IList<T> ReadObjectAsList<T>(IUtf8JsonStreamReaderConverter<T> streamReaderConverter)
+        {
+            if (TokenType != JsonTokenType.StartObject)
+            {
+                return Array.Empty<T>();
```

Comment:
If this was `if (TokenType == JsonTokenType.Null) { return Array.Empty<T>(); }`, I'd be ok with it.

But the way this code is written makes me think that json `{ "propertyName": 4 }` would return the empty array, rather than an exception saying the JSON is invalid.